### PR TITLE
123 update spring boot models

### PR DESCRIPTION
**Pull request by PaulPilane**
# Description

Updated Spring Boot Models. Implemented Profile, Review, Reference, Meal Plan, Recipe Controllers.for saving to and retrieving data from the database

Fixes # (issue)
#123 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/api/src/main/java/com/fridgetoplate/response/RecipeResponse.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTyped;
+import com.fridgetoplate.interfaces.Recipe;
+import com.fridgetoplate.model.Ingredient;
+import com.fridgetoplate.model.Review;
+
+
+public class RecipeResponse extends Recipe {
+
+    private List<Review> reviews;
+    private Integer views = 0;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/response/RecipeResponse.java:18]: The response shouldn't send the views

File: apps/api/src/main/java/com/fridgetoplate/model/Ingredient.java [hunk 2]
```diff
@@ -3,17 +3,28 @@
...
 import lombok.Data;
 
 @Data
 @DynamoDBTable(tableName = "ingredients")
 public class Ingredient {
-    @DynamoDBHashKey
+    
+    @DynamoDBHashKey(attributeName = "ingredientId")
     @DynamoDBAutoGeneratedKey
     private String ingredientId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/Ingredient.java:18]: There isn't an ingredientID

File: apps/api/src/main/java/com/fridgetoplate/model/Preference.java [hunk 3]
```diff
@@ -10,10 +10,20 @@
 @DynamoDBTable(tableName = "preferences")
 public class Preference {
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/Preference.java:11]: Rename class to Preferences

File: apps/api/src/main/java/com/fridgetoplate/interfaces/Profile.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.List;
+
+import com.fridgetoplate.model.Ingredient;
+
+import lombok.Data;
+
+@Data
+public class Profile {
+
+    protected String profileId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/interfaces/Profile.java:11]: There isn't a profileId

File: apps/api/src/main/java/com/fridgetoplate/interfaces/Recipe.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+public class Recipe extends RecipeDesc {
+
+    protected String description;
+
+
+    protected String meal;
+
+    protected Integer prepTime;
+
+    protected Integer numberOfServings;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/interfaces/Recipe.java:15]: Please conform to recipe model

File: apps/api/src/main/java/com/fridgetoplate/interfaces/Recipe.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    protected List<Ingredient> ingredients;
+
+    protected List<String> instructions;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/interfaces/Recipe.java:19]: Please conform to recipe model

File: apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+    private String breakfastId;
+
+    private String lunchId;
+    
+    private String dinnerId;
+
+    private String snackId;
+
+    private String dessertId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java:21]: Please conform to meal plan model

File: apps/api/src/main/java/com/fridgetoplate/model/RecipeModel.java [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAutoGeneratedKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperFieldModel;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTyped;
+import com.fridgetoplate.interfaces.Recipe;
+
+@DynamoDBTable(tableName = "recipes")
+public class RecipeModel extends Recipe {
+
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/RecipeModel.java:14]: Should be views attribute here

File: apps/api/src/main/java/com/fridgetoplate/model/Review.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+
+import lombok.Data;
+@Data
+@DynamoDBTable(tableName = "reviews")
+public class Review {
+
+    @DynamoDBHashKey(attributeName = "recipeId")
+    private String recipeId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/Review.java:15]: Please conform to reviews model

File: apps/api/src/main/java/com/fridgetoplate/model/Review.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+    @DynamoDBRangeKey(attributeName = "reviewId")
+    @DynamoDBAutoGeneratedKey
+    private String reviewId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/Review.java:19]: Please conform to reviews model

File: apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.fridgetoplate.interfaces.MealPlan;
+
+@DynamoDBTable(tableName = "meal_plans")
+public class MealPlanModel extends MealPlan {
+
+
+    private String breakfastId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java:13]: Please conform to meal plan model model

File: apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+    private String lunchId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java:15]: Please conform to meal plan model model

File: apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+    
+    private String dinnerId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java:17]: Please conform to meal plan model model

File: apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+    private String snackId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/model/MealPlanModel.java:19]: Please conform to meal plan model model

File: apps/api/src/main/java/com/fridgetoplate/interfaces/RecipeDesc.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
+package com.fridgetoplate.interfaces;
+
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class RecipeDesc {
+
+    protected String recipeId;
```
Comment (ryanbasiltrickett) [inline apps/api/src/main/java/com/fridgetoplate/interfaces/RecipeDesc.java:9]: Please conform to recipe description model

**Review by ryanbasiltrickett — CHANGES_REQUESTED** [review]
Please conform to Spring models, and complete your PR labels, milestone, etc

File: apps/api/src/main/java/com/fridgetoplate/response/RecipeResponse.java (hunk 1 above)
Comment (PaulPilane) [reply apps/api/src/main/java/com/fridgetoplate/response/RecipeResponse.java:18]: What is the point of having views then if they will not be shown?

File: apps/api/src/main/java/com/fridgetoplate/model/Ingredient.java (hunk 2 above)
Comment (PaulPilane) [reply apps/api/src/main/java/com/fridgetoplate/model/Ingredient.java:18]: Alright, the names shall be their unique id

File: apps/api/src/main/java/com/fridgetoplate/interfaces/Profile.java (hunk 4 above)
Comment (PaulPilane) [reply apps/api/src/main/java/com/fridgetoplate/interfaces/Profile.java:11]: Noted

File: apps/api/src/main/java/com/fridgetoplate/response/RecipeResponse.java (hunk 1 above)
Comment (ryanbasiltrickett) [reply apps/api/src/main/java/com/fridgetoplate/response/RecipeResponse.java:18]: For backend analytics analysis

**Review by ryanbasiltrickett — APPROVED** [review]

**Pull request merged**